Tests/UnboxTests.swift

```diff
@@ -359,6 +359,138 @@ class UnboxTests: XCTestCase {
         }
     }
     
+    func testCustomDictionaryKeyTypeWithArrayOfUnboxables() {
+        struct Model: Unboxable {
+            let requiredModelDictionary: [UnboxTestDictionaryKey : [UnboxTestSimpleMock]]
+            let optionalModelDictionary: [UnboxTestDictionaryKey : [UnboxTestSimpleMock]]?
+            
+            init(unboxer: Unboxer) {
+                self.requiredModelDictionary = unboxer.unbox("requiredModelDictionary")
+                self.optionalModelDictionary = unboxer.unbox("optionalModelDictionary")
+            }
+        }
+        
+        do {
+            let unboxed: Model = try Unbox([
+                "requiredModelDictionary" : [
+                    "key" : [
+                        ["int" : 31]
+                    ]
+                ],
+                "optionalModelDictionary" : [
+                    "optionalKey" : [
+                        ["int" : 19]
+                    ]
+                ]
+                ])
+            
+            if let values = unboxed.requiredModelDictionary[UnboxTestDictionaryKey(key: "key")] {
+                XCTAssertEqual(values, [UnboxTestSimpleMock(int: 31)])
+            } else {
+                XCTFail("Key was missing from unboxed dictionary")
+            }
+            
+            if let values = unboxed.optionalModelDictionary?[UnboxTestDictionaryKey(key: "optionalKey")] {
+                XCTAssertEqual(values, [UnboxTestSimpleMock(int: 19)])
+            } else {
+                XCTFail("Key was missing from unboxed dictionary")
+            }
+            
+            let unboxedWithoutOptionals: Model = try Unbox([
+                "requiredModelDictionary" : [
+                    "key" : [
+                        ["int" : 31]
+                    ]
+                ]
+                ])
```

Comment:
Incorrect indentation here (and in a few other places with the same formatting).